client/my-sites/promote-post-i2/components/posts-list/header.tsx

```diff
@@ -41,10 +45,56 @@ export default function PostsListHeader() {
 		},
 	];
 
+	const productColumns: Array< PostsListHeaderColumn > = [
+		{
+			id: 'data',
+			title: translate( 'Post' ),
+		},
+		{
+			id: 'type',
+			title: translate( 'Type' ),
+		},
+		{
+			id: 'sku',
+			title: translate( 'SKU' ),
+		},
+		{
+			id: 'price',
+			title: translate( 'Price' ),
+		},
+		{
+			id: 'publish-date',
+			title: translate( 'Publish date' ),
+		},
+		{
+			id: 'visitors',
+			title: translate( 'Visitors' ),
+		},
+		{
+			id: 'view',
+			title: null,
+		},
+		{
+			id: 'promote',
+			title: null,
+		},
+	];
+
+	const getHeaderColumns = (): Array< PostsListHeaderColumn > => {
```

Comment:
The changes between the columns are very minimal (just hiding two columns and showing other two. 
Instead of duplicating the full header list, maybe it will be better to generate the list in this function, and add some dynamic check to know what columns to apply.

For example, we can have something like this:
```

const postColumns: Array< PostsListHeaderColumn > = [
  {
    id: 'data',
	title: translate( 'Post' ),
  }, 
  ...
];

if (postType === 'product') {
  postColumns.push({...})
}

postColumns.push({
  id: 'views',
  title: translate( 'Visitors' ),
});

if (postType !== 'product') {
  postColumns.push({...})
}

...

return postColumns;
```

This is just an idea, there are multiple ways to do it. Using something like this will make us reduce duplication, and avoid future mistakes if anyone changes the label of a shared column.

Reply:
Fixed